core/ledger/ledger_interface.go

```diff
@@ -733,9 +733,19 @@ type HashProvider interface {
 	GetHash(opts bccsp.HashOpts) (hash.Hash, error)
 }
 
+// CommitNotification is sent on each block commit to the channel returned by PeerLedger.CommitNotificationsChannel()
 type CommitNotification struct {
-	BlockNumber         uint64
-	TxIDValidationCodes map[string]peer.TxValidationCode
+	BlockNumber uint64
+	TxsByTxID   map[string]*CommitNotificationTxInfo
+}
+
+// CommitNotificationTxInfo contains the details of a transaction that is included in the CommitNotification
+// ChaincodeID will be nil if the transaction is not an endorser transaction. This may or may not be nil if the tranasction is invalid. However, it will always be non-nil if the transaction is valid.
```

Comment:
Can you expand on "This may or may not be nil if the transaction is invalid".  What are the different scenarios?

Reply:
Added more details in the comment